backend/src/main/java/org/eclipse/tractusx/puris/backend/common/ddtr/logic/util/DtrRequestBodyBuilder.java

```diff
@@ -0,0 +1,252 @@
+        var refObject = objectMapper.createObjectNode();
+        refObject.put("type", "GlobalReference");
+        refObject.put("value", bpnl);
+        return refObject;
+    }
+
+    private ObjectNode getOwnReferenceObject() {
+        return getReferenceObject(variablesService.getOwnBpnl());
+    }
+
+
+    private JsonNode getDigitalTwinObject(List<ObjectNode> refNodes) {
+        var digitalTwinObject = objectMapper.createObjectNode();
+        digitalTwinObject.put("name", DIGITAL_TWIN_TYPE);
+        digitalTwinObject.put("value", "PartType");
+        var externalSubjectIdObject = objectMapper.createObjectNode();
+        digitalTwinObject.set("externalSubjectId", externalSubjectIdObject);
+        externalSubjectIdObject.put("type", "ExternalReference");
+        var keysArray = objectMapper.createArrayNode();
+        externalSubjectIdObject.set("keys", keysArray);
+        keysArray.add(getOwnReferenceObject());
+        for (var refNode : refNodes) {
+            keysArray.add(refNode);
+        }
+        return digitalTwinObject;
+    }
+
+    private JsonNode getItemStockSubmodelObject() {
+        var itemStockRequestSubmodelObject = objectMapper.createObjectNode();
+
+        itemStockRequestSubmodelObject.put("id", UUID.randomUUID().toString());
+        var semanticIdObject = objectMapper.createObjectNode();
+        itemStockRequestSubmodelObject.set("semanticId", semanticIdObject);
+        semanticIdObject.put("type", "ExternalReference");
+        var keysArray = objectMapper.createArrayNode();
+        semanticIdObject.set("keys", keysArray);
+        var keyObject = objectMapper.createObjectNode();
+        keysArray.add(keyObject);
+        keyObject.put("type", "GlobalReference");
+        keyObject.put("value", "urn:samm:io.catenax.item_stock:2.0.0#ItemStock");
+
+        var endpointsArray = objectMapper.createArrayNode();
+        itemStockRequestSubmodelObject.set("endpoints", endpointsArray);
+        var submodel3EndpointObject = objectMapper.createObjectNode();
+        endpointsArray.add(submodel3EndpointObject);
+        submodel3EndpointObject.put("interface", "SUBMODEL-3.0");
+        var protocolInformationObject = objectMapper.createObjectNode();
+        submodel3EndpointObject.set("protocolInformation", protocolInformationObject);
+        protocolInformationObject.put("href", variablesService.getEdcProtocolUrl());
+        protocolInformationObject.put("endpointProtocol", "HTTP");
+        var endpointProtocolVersionArray = objectMapper.createArrayNode();
+        protocolInformationObject.set("endpointProtocolVersion", endpointProtocolVersionArray);
+        endpointProtocolVersionArray.add("1.1");
+        protocolInformationObject.put("subprotocol", "DSP");
+        protocolInformationObject.put("subprotocolBodyEncoding", "plain");
+        protocolInformationObject.put("subprotocolBody", "id=FOO;dspEndpoint=" + variablesService.getEdcProtocolUrl());
```

Comment:
Well, the problem is that there is no single EDC-asset id for the entire ItemStockSubmodel. And there are good reasons why there should not be a single asset id. 

At best, we could arbitrarily insert one of the many asset ids here. But that would not be sensible either. Since no partner could actually rely on that id. 

There is an established and sensible workflow, how to identify the correct asset from a partner's EDC catalog via the EDC-asset properties of the ItemStockSubmodel. An id is frankly not needed to find the ItemStock API asset in the catalog.